mne/preprocessing/tests/test_realign.py

```diff
@@ -75,60 +74,104 @@ def test_realign(ratio_other, start_raw, start_other, stop_raw, stop_other, test
     t_other = (events_other[:, 0] - other.first_samp) / other.info["sfreq"]
     assert duration - 10 <= len(events_raw) < duration
     raw_orig, other_orig = raw.copy(), other.copy()
-    realign_raw(raw, other, t_raw, t_other, realign_annot=test_annot)
+    realign_raw(raw, other, t_raw, t_other)
 
-    # old events should still work for raw and produce the same result
+    # old events should still work for raw and produce the same evoked data
     evoked_raw_2, events_raw, _, events_other = _assert_similarity(
-        raw, other, n_events, events_raw=events_raw
+        raw, other, n_events, ratio_other, events_raw=events_raw
     )
     assert_allclose(evoked_raw.data, evoked_raw_2.data)
     assert_allclose(raw.times, other.times)
 
     # raw data now aligned
-    hann_corr = np.corrcoef(raw.get_data([1])[0], other.get_data([1])[0])[0, 1]
-    assert 0.98 < hann_corr <= 1.0
-    box_corr = np.corrcoef(raw.get_data([2])[0], other.get_data([2])[0])[0, 1]
-    assert 0.98 < box_corr <= 1.0
-
-    # annotations now aligned
-    if test_annot:
-        assert_allclose(
-            raw.annotations.onset, events_raw[:, 0] / raw.info["sfreq"], atol=1 / sfreq
-        )
-        assert_allclose(
-            other.annotations.onset,
-            events_other[:, 0] / other.info["sfreq"],
-            atol=1 / sfreq,
-        )
-        onsets_raw = raw.annotations.onset - raw.first_samp / sfreq
-        dur_raw = raw.annotations.duration
-        onsets_other = other.annotations.onset - other.first_samp / sfreq
-        dur_other = other.annotations.duration
-        assert_allclose(onsets_raw, onsets_other, atol=1 / sfreq)
-        assert_allclose(dur_raw, dur_other, atol=1 / sfreq)
+    corr = np.corrcoef(raw.get_data("data"), other.get_data("data"))
+    assert 0.98 < corr[0, 2] <= 1.0  # hanning
+    assert 0.98 < corr[1, 3] <= 1.0  # boxcar
+
+    # onsets drived from stim and annotations are the same
+    assert_allclose(
+        raw.annotations.onset, events_raw[:, 0] / raw.info["sfreq"], atol=2 / sfreq
+    )
+    assert_allclose(
+        other.annotations.onset,
+        events_other[:, 0] / other.info["sfreq"],
+        atol=2 / sfreq,
+    )
+
+    # onsets (relative to first sample) and durations
+    first_time_raw = raw.first_samp / raw.info["sfreq"]
+    onsets_raw = raw.annotations.onset - first_time_raw
+    dur_raw = raw.annotations.duration
+
+    first_time_other = other.first_samp / other.info["sfreq"]
+    onsets_other = other.annotations.onset - first_time_other
+    dur_other = other.annotations.duration
+
+    # onsets and durations now aligned
+    assert len(onsets_raw) == len(onsets_other) == len(events_raw)
+    assert_allclose(onsets_raw, onsets_other, atol=2 / sfreq)
+    assert_allclose(dur_raw, dur_other, atol=2 / sfreq)
+
+    # onset/offset correspond to 0/1 transition in boxcar signals
+    n_events = len(onsets_raw)
+    onsets_samp_raw = (onsets_raw * sfreq).astype(int)
```

Comment:
safer here to to `raw.time_as_index(onsets_raw)` to get the sample indices. That should also allow you to shrink the 10-sample window down to something much smaller.

Reply:
This is very good suggestion! It does shrink the sample window as well!